### PR TITLE
Feat: get tx hashes for address

### PR DESCRIPTION
**Pull request by josepfo**

### CODE REVIEW

File: lib/plugins/wallet/bitcoincashd/bitcoincashd.js [hunk 1]
```diff
@@ -129,12 +129,20 @@ function checkBlockchainStatus (cryptoCode) {
     .then(res => !!res['initialblockdownload'] ? 'syncing' : 'ready')
 }
 
+function getTxHashesByAddress (cryptoCode, address) {
+  checkCryptoCode(cryptoCode)
+    .then(() => fetch('listreceivedbyaddress', [0, true, true, address]))
+    .then(txsByAddress => Promise.all(_.forEach(id => fetch('getrawtransaction', [id]), _.head(txsByAddress).txids)))
```
Comment (chaotixkilla) [inline lib/plugins/wallet/bitcoincashd/bitcoincashd.js]: As we spoke on PMs

```suggestion
    .then(txsByAddress => Promise.all(_.map(id => fetch('getrawtransaction', [id]), _.flatMap(it => it.txids, txsByAddress))))
```

File: lib/plugins/wallet/bitcoind/bitcoind.js [hunk 2]
```diff
@@ -191,6 +191,13 @@ function checkBlockchainStatus (cryptoCode) {
     .then(res => !!res['initialblockdownload'] ? 'syncing' : 'ready')
 }
 
+function getTxHashesByAddress (cryptoCode, address) {
+  checkCryptoCode(cryptoCode)
+    .then(() => fetch('listreceivedbyaddress', [0, true, true, address]))
+    .then(txsByAddress => Promise.all(_.forEach(id => fetch('getrawtransaction', [id]), _.head(txsByAddress).txids)))
```
Comment (chaotixkilla) [inline lib/plugins/wallet/bitcoind/bitcoind.js]: Same here

File: lib/plugins/wallet/dashd/dashd.js [hunk 3]
```diff
@@ -125,11 +125,19 @@ function checkBlockchainStatus (cryptoCode) {
     .then(res => !!res['initialblockdownload'] ? 'syncing' : 'ready')
 }
 
+function getTxHashesByAddress (cryptoCode, address) {
+  checkCryptoCode(cryptoCode)
+    .then(() => fetch('listreceivedbyaddress', [0, true, true, true, address]))
+    .then(txsByAddress => Promise.all(_.forEach(id => fetch('getrawtransaction', [id]), _.head(txsByAddress).txids)))
```
Comment (chaotixkilla) [inline lib/plugins/wallet/dashd/dashd.js]: Here

File: lib/plugins/wallet/litecoind/litecoind.js [hunk 4]
```diff
@@ -125,11 +125,16 @@ function checkBlockchainStatus (cryptoCode) {
     .then(res => !!res['initialblockdownload'] ? 'syncing' : 'ready')
 }
 
+function getTxHashesByAddress (cryptoCode, address) {
+  throw new Error(`Transactions hash retrieval not implemented for this coin!`)
+}
+
```
Comment (chaotixkilla) [inline lib/plugins/wallet/litecoind/litecoind.js:131]: Litecoin in theory has access to similar RPC methods as Bitcoin, should be the same implementation

```suggestion
function getTxHashesByAddress (cryptoCode, address) {
  checkCryptoCode(cryptoCode)
    .then(() => fetch('listreceivedbyaddress', [0, true, true, address]))
    .then(txsByAddress => Promise.all(_.map(id => fetch('getrawtransaction', [id]), _.flatMap(it => it.txids, txsByAddress))))
    .then(_.map(({ hash }) => hash))
}

```

File: lib/plugins/wallet/zcashd/zcashd.js [hunk 5]
```diff
@@ -151,11 +151,18 @@ function checkBlockchainStatus (cryptoCode) {
     .then(res => !!res['initial_block_download_complete'] ? 'ready' : 'syncing')
 }
 
+function getTxHashesByAddress (cryptoCode, address) {
+  checkCryptoCode(cryptoCode)
+    .then(() => fetch('getaddresstxids', [address]))
+    .then(_.values)
+}
```
Comment (chaotixkilla) [inline lib/plugins/wallet/zcashd/zcashd.js]: ZEC 5.1.0 documentation shows that the `getaddresstxids` already returns an array with the transaction IDs, so there shouldn't be need for `_.values` https://zcash.github.io/rpc/getaddresstxids.html

```suggestion
function getTxHashesByAddress (cryptoCode, address) {
  checkCryptoCode(cryptoCode)
    .then(() => fetch('getaddresstxids', [address]))
}
```

File: lib/plugins/wallet/monerod/monerod.js [hunk 6]
```diff
@@ -235,12 +235,21 @@ function checkBlockchainStatus (cryptoCode) {
   })
 }
 
+function getTxHashesByAddress (cryptoCode, address) {
+  checkCryptoCode(cryptoCode)
+    .then(() => refreshWallet())
+    .then(() => fetch('get_address_index', { address: address }))
+    .then(addressRes => fetch('get_transfers', { in: true, pool: true, pending: true, account_index: addressRes.index.major, address_indices: [addressRes.index.minor] }))
```
Comment (chaotixkilla) [inline lib/plugins/wallet/monerod/monerod.js]: One of the parameters is wrong, according to https://www.getmonero.org/resources/developer-guides/wallet-rpc.html#get_transfers

```suggestion
    .then(addressRes => fetch('get_transfers', { in: true, pool: true, pending: true, account_index: addressRes.index.major, subaddr_indices: [addressRes.index.minor] }))
```

**Review by chaotixkilla — CHANGES_REQUESTED** [review]

File: lib/plugins/wallet/monerod/monerod.js (hunk 6 above)
Comment (josepfo) [reply lib/plugins/wallet/monerod/monerod.js]: Good catch! I'll change this on other `get_transfers` calls as well.

**Review by chaotixkilla — APPROVED** [review]

**Review by chaotixkilla — APPROVED** [review]

**Pull request merged**